compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -391,22 +412,25 @@ private class SyntheticAccessorLowering(val context: JvmBackendContext) : IrElem
         }
     }
 
+    // Counter used to disambiguate accessors to functions with the same base name.
+    private var nameCounter = 0
+
     private fun IrFunction.accessorName(): Name {
         val jvmName = context.state.typeMapper.mapFunctionName(
             descriptor,
             OwnerKind.getMemberOwnerKind(parentAsClass.descriptor)
         )
-        return Name.identifier("access\$$jvmName")
+        return Name.identifier("access\$$jvmName\$${nameCounter++}")
```

Comment:
@gogabr @sfs Could we use old naming scheme from current backend? ('getAccessorNameSuffix' in syntheticAccessorUtil.kt) This will allow to unmute a bunch of bytecode tests.

Reply:
If yes it could be postponed for another PR

Reply:
I considered this, but right now there's a bug in the old naming scheme which can lead to name clashes (reported as [KT-32870](https://youtrack.jetbrains.com/issue/KT-32870)). Is this naming scheme part of any public ABI?

However, I didn't see that there were tests which depend on the naming scheme. In that case it might be better to use it now and fix it later. I'll have a look.

Reply:
There are some tests, e.g: bytecodeText/companion, bytecodeText/jvm8/jvmDefault/compatibility.

It's a part of public api in a sence that inlined code wouldn't work after binary library update. Change in naming scheme should be accepted by Language Committee.
Seems as code generated by IR backend should be binary compatible with old one we should support old scheme in any way,  should it @udalov?

Reply:
It turns out that this is not straightforward to implement.

For one, the naming scheme depends on the origin of fields (e.g., moved from companion classes, generated for locals) which is not recorded in the ir backend. For another, the "super call" suffix also needs to be added from field accesses (IrGetField, IrSetField), which currently don't record a `superQualifierSymbol` for static fields. The latter ends up breaking `testProtectedStatic2`, which was why I had to put in the counter in the first place.

If possible, I'd like to postpone this change.

Reply:
@sfs Actually protected member calls from public inline functions would be prohibited in 1.4 (https://youtrack.jetbrains.com/issue/KT-21178). So most of inline cases around synthetic accessors shouldn't be considered as public ABI, except ones with @PublishedApi internal functions

Reply:
Update comment above^